functionary/ui/forms/scheduled_task.py

```diff
@@ -11,7 +13,26 @@
 from core.models import Environment, Function, ScheduledTask
 
 
+class NotNoneJSONField(JSONField):
```

Comment:
I see how you arrived at this and I'll admit that the behavior of the form validation and handling of the empty value for parameters threw me for a loop when trying to understand what's going on here.  But this definitely feels like overkill and I don't think we should need to resort to this.

I tried playing with the `ScheduledTask` and the validation process to see if I could fix this more simply, and I think maybe doing the following gets us the behavior that we want:

* Updated the `ScheduledTask` parameters field to have `blank=True, default=dict`.  The blank part allows the form validation to pass when there are no parameters and the default means that when the instance is saved the parameters will be `{}` instead of `None`.
* Handle parameters=None in the `validate_parameters` util function:
   ```python
    if parameters is None:
        parameters = {}
    ```

That feels like a more natural solution than what's going on here.  If there are other scenarios that your solution is covering that what I've described above would not let me know.  But I'm hopeful that worst case is we could iterate on what's above and get there.